### DISCUSSION TITLE
Deprecate ma.INFINITY and add ma.MAX

### DISCUSSION THREAD

**Discussion by dariosanfilippo** [grame-cncm/faustlibraries#94] [Ideas]
Hello, people.

After some discussion on the topic, I was thinking that it could be appropriate to deprecate ma.INFINITY and add ma.MAX with the same definition. This is because ma.INFINITY and the actual INF special value behave differently and it can be inconsistent. For example: ma.INFINITY - ma.INFINITY = 0 while inf - inf = NaN.

Ciao,
Dario

**Comment by sletz** [comment]
- do you mean `ma.INFINITY` was incorrectly named and should be renamed as `ma.MAX` ? (see https://docs.microsoft.com/en-us/cpp/c-language/limits-on-floating-point-constants?view=msvc-160)
-  and what do you mean by _the actual INF special_  ?

**Reply by dariosanfilippo** [reply]
I don't think that we can remove it as I can think of some code using it, for example, Bart's limiter, if I'm not wrong, but we could add ma.MAX as the corresponding FP MAX constant.

I'm distinguishing between the INF generated by, say, 1.0/0.0 and the constant ma.INFINITY which is actually a number.

**Reply by magnetophon** [reply]
> I don't think that we can remove it as I can think of some code using it, for example, Bart's limiter, if I'm not wrong, but we could add ma.MAX as the corresponding FP MAX constant.

I am indeed using ``ma.INFINITY`` in https://github.com/grame-cncm/faustlibraries/blob/master/basics.lib

Should those be search-replaced with ``ma.MAX``?

**Reply by sletz** [reply]
Yes better, although `ma.INFINITY` is maintained for compatibility reasons.

**Comment by sletz** [comment]
OK, done https://github.com/grame-cncm/faustlibraries/commit/358a80472d15416c00e7048844526aeabf5c6f87.
